pkgs/os-specific/linux/systemd/vmlinux-btf.nix

```diff
@@ -0,0 +1,59 @@
+{
+  lib,
+  linuxManualConfig,
+  linuxConfig,
+  linuxHeaders,
+  bpftools,
+  runCommand,
+}:
+let
+  # Create a minimal kernel config file, which is sufficient for the BTF
+  # that systemd uses.
+  configfile =
+    (linuxConfig {
+      inherit (linuxHeaders) src;
+      makeTarget = "allnoconfig";
+    }).overrideAttrs
+      (oldAttrs: {
+        buildPhase =
+          oldAttrs.buildPhase
```

Comment:
This is mostly curiosity, but could this have been done by providing `extraSrtructuredConfig` via `kernelPatches` rather than overriding the build phase like this?

Reply:
It's very unfortunate, and I have tried each and every other option to make this work, but `linuxConfig` doesn't have a convenient way to pass patches to the kernel config like `buildLinux` and friends have.